File: 3075.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        res = 0
        for i in range(k):
            diff = happiness[i] - i
            if diff > 0:
                res += diff
            else:
                break
        return res

File: sol/3075.cpp
struct Solution {
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        ranges::sort(happiness, std::greater{});
        long long res = 0;
        int diff;
        for (int i = 0; i < k; ++i){
            diff = happiness[i] - i;
            if (diff > 0){
                res += diff;
            } else {
                break;
            }
        }
        return res;
    }
};
